### ISSUE TITLE
Equation and boundary condition

### ISSUE THREAD

**Issue by lecopivo** [m-ender/webgl-ripples#2]
I have a question about the equation you used.

All the waves travels downward so I guess you added a convection term into the wave equation, to make this happen. But what boggles me is, what are the boundary conditions? It seams to me, that the waves does not bounce of the edge of the screen, I find this very strange. 

What equation exactly do you calculate and what are the boundary conditions?

Thank you,
Tomas

**Comment by m-ender** [comment]
Sorry for the delay in replying. I'm not sure what you mean by all the waves travelling downward. That shouldn't be the case - they should radiate out uniformly from their epicentre. So no, there is no convection term. The equation I've used is really just the most basic [wave equation](https://en.wikipedia.org/wiki/Wave_equation#Introduction).

As for the boundary conditions, the waves should definitely bounce back (from the edges of the viewport and the letters). However, I don't recall if I actually implemented the b.c.s as solid walls (inverting the waves which travel back) or as free b.c.s (such that the reflected wave has the same phase and you get constructive interference at the boundary). Judging by eye, the latter seems more likely (and I know that I definitely had those at some point, but I don't recall if I kept them in the final version).

It sounds like the code simply isn't working as it should on your machine (which isn't too surprising, because I never tested this anywhere except on Windows using Chrome). For reference, here is a screenshot of what the simulation looks like on my machine:

![ripples](https://cloud.githubusercontent.com/assets/1846708/10223522/a1d78030-6859-11e5-87f2-df38d7b4ac6f.png)

**Comment by lecopivo** [comment]
Yeah, it is not working on my machine. That is unfortunate, I hoped that you
implemented non-reflecting boundary. I was thinking quite a lot about non-
reflecting bc but I couldnt come up with anything. Anyway thanks for a 
reply.

Best regards,

Tomas

---------- Původní zpráva ----------

Od: Martin Büttner 

Datum: 1. 10. 2015 v 16:32:42

Předmět: Re: [webgl-ripples] Equation and boundary condition (#2)

Sorry for the delay in replying. I'm not sure what you mean by all the waves
travelling downward. That shouldn't be the case - they should radiate out 
uniformly from their epicentre. So no, there is no convection term. The 
equation I've used is really just the most basic wave equation
(https://en.wikipedia.org/wiki/Wave_equation#Introduction).

As for the boundary conditions, the waves should definitely bounce back 
(from the edges of the viewport and the letters). However, I don't recall if
I actually implemented the b.c.s as solid walls (inverting the waves which 
travel back) or as free b.c.s (such that the reflected wave has the same 
phase and you get constructive interference at the boundary). Judging by 
eye, the latter seems more likely (and I know that I definitely had those at
some point, but I don't recall if I kept them in the final version).

It sounds like the code simply isn't working as it should on your machine 
(which isn't too surprising, because I never tested this anywhere except on 
Windows using Chrome). For reference, here is a screenshot of what the 
simulation looks like on my machine:

ripples 
(https://cloud.githubusercontent.com/assets/1846708/10223522/a1d78030-6859-11e5-87f2-df38d7b4ac6f.png)

—

Reply to this email directly or view it on GitHub
(https://github.com/mbuettner/webgl-ripples/issues/2#issuecomment-144746184)
. 

#

**Comment by willsolve** [comment]
The downward effect appears to happen during an initial load in Chrome or IE.  Upon reload with the cached files, it works fine.  It's quite an interesting effect:

![image](https://cloud.githubusercontent.com/assets/1574051/15172265/be2cda96-1709-11e6-8c0b-2fb09532effa.png)
